### PR TITLE
News Article Parser

### PR DESCRIPTION
**Pull request by ankitshu**
Parses news articles using the News API. The bot extracts title, author, description, and URL from articles based on user-provided keywords. Added error handling for cases where no articles are found.

### CODE REVIEW

**Review by ankitshu — COMMENTED** [review]
API key change with property

**Review by ashu759 — COMMENTED** [review]
Can you add readme file as well?

**Comment by ankitshu** [comment]
@ashu759  Shall i add that in the Parser as per the contribution file it was not needed and i don't see any other Read me there , Just a quick confirmation

File: Parsers/News Article Parser.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+activation_example:!parse_news name of news website
```
Comment (chelming) [inline Parsers/News Article Parser.js:2]: should this be "name of news website"  or "topic or story to search for"?

File: Parsers/News Article Parser.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+regex:!parse_news
+flags:gmi
+*/
+
+var where = current.text.indexOf('!parse_news ') + 12;
+var term = current.text.substr(where).trim();
```
Comment (chelming) [inline Parsers/News Article Parser.js:8]: I'd just change this to a `.replace("!news ", "")` and update the regex at the top to `!news`

File: Parsers/News Article Parser.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+activation_example:!parse_news name of news website
+regex:!parse_news
```
Comment (chelming) [inline Parsers/News Article Parser.js:3]: I think `!news` makes more sense (and is easier to type)

File: Parsers/News Article Parser.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// Parse the API response as JSON
+var responseData = JSON.parse(chatResponseBody);
+
+// Check if the API response contains any articles
+if (responseData.articles.length > 0) {
+  // Extract information from the first article
+  var article = responseData.articles[0];
+  var title = article.title;
+  var author = article.author;
+  var description = article.description;
```
Comment (chelming) [inline Parsers/News Article Parser.js:29]: I think you should trim this at something like 255 chars and add `+ "... Read More"` so it doesn't spam a channel with a large amount of text.

File: Parsers/News Article Parser.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  var url = article.url;
+
+  // Send the extracted information as a Slack message
+  var message = "**Title:** " + title + "\n**Author:** " + author + "\n**Description:** " + description + "\n**URL:** " + url;
```
Comment (chelming) [inline Parsers/News Article Parser.js:33]: Slack markdown uses single `*`s for `*bold*` text.

File: Parsers/News Article Parser.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  new x_snc_slackerbot.Slacker().send_chat(current, message, false);
```
Comment (chelming) [inline Parsers/News Article Parser.js:34]: change that third param from false to true so it threads the response. there's quite a bit of text here so it'd be better to keep it out of a main channel.

**Review by chelming — CHANGES_REQUESTED** [review]

File: Parsers/News Article Parser.js (hunk 1 above)
Comment (ankitshu) [reply Parsers/News Article Parser.js]: @chelming This has been addressed in the new commit

File: Parsers/News Article Parser.js (hunk 2 above)
Comment (ankitshu) [reply Parsers/News Article Parser.js]: @chelming This has been addressed in the new commit

File: Parsers/News Article Parser.js (hunk 4 above)
Comment (ankitshu) [reply Parsers/News Article Parser.js]: @chelming This has been addressed Thanks for the Suggestion

File: Parsers/News Article Parser.js (hunk 5 above)
Comment (ankitshu) [reply Parsers/News Article Parser.js]: @chelming This has been addressed

File: Parsers/News Article Parser.js (hunk 6 above)
Comment (ankitshu) [reply Parsers/News Article Parser.js]: @chelming This has been addressed

**Pull request merged**